### PR TITLE
Fix Unique Keys not being updated when changing a user's email address

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?
This PR Fixes the behavior of not being able to create a new account with a email address if it has been previously used but changed using the change email endpoint. It fixes this by correctly cleaning up the unique key when the email gets updated.

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/1104
https://github.com/appwrite/appwrite/issues/1295

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes, I have

### CODE REVIEW

File: app/controllers/api/account.php [hunk 1]
```diff
@@ -1103,14 +1103,23 @@
...
             $user->getArrayCopy(),
             ($isAnonymousUser ? [ 'password' => Auth::passwordHash($password) ] : []),
             [
                 'email' => $email,
                 'emailVerification' => false,
             ]
-        ));
+            ));
+
+        $user = $projectDB->updateDocument($document, ['email' => $email]);
```
Comment (eldadfux) [inline app/controllers/api/account.php:1120]: As far as I checked `updateDocument` accepts only 1 param.

Comment (PineappleIOnic) [reply app/controllers/api/account.php:1120]: Yep, I probably missed that out as my original version added a unique param to updateDocument :sweat_smile: I'll fix it now

File: src/Appwrite/Database/Adapter/MySQL.php [hunk 2]
```diff
@@ -377,6 +377,31 @@ public function deleteUniqueKey($key)
         return [];
     }
 
+    /**
+     * Add Unique Key.
+     *
+     * @param String $key
```
Comment (TorstenDittmann) [inline src/Appwrite/Database/Adapter/MySQL.php:383]: primitive data types are lowercase.

File: src/Appwrite/Database/Adapter/MySQL.php [hunk 3]
```diff
@@ -360,7 +360,7 @@ public function deleteDocument(string $id)
     /**
      * Delete Unique Key.
      *
-     * @param int $key
+     * @param String $key
```
Comment (TorstenDittmann) [inline src/Appwrite/Database/Adapter/MySQL.php:363]: primitive data types are lowercase.

File: src/Appwrite/Database/Adapter.php [hunk 4]
```diff
@@ -91,12 +91,21 @@ abstract public function deleteDocument(string $id);
...
+     * @param String $key
      *
      * @return array
      */
     abstract public function deleteUniqueKey($key);
 
+    /**
+     * Add Unique Key.
+     *
+     * @param String $key
```
Comment (TorstenDittmann) [inline src/Appwrite/Database/Adapter.php:103]: primitive data types are lowercase.

File: src/Appwrite/Database/Adapter.php [hunk 5]
```diff
@@ -91,12 +91,21 @@ abstract public function deleteDocument(string $id);
     /**
      * Delete Unique Key.
      *
-     * @param int $key
+     * @param String $key
```
Comment (TorstenDittmann) [inline src/Appwrite/Database/Adapter.php:94]: primitive data types are lowercase.

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]

**Review by TorstenDittmann — COMMENTED** [review]
Can we extend the test where the e-mail is updated and we create an account with the old email after?

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**